crates/red_knot_python_semantic/src/types.rs

```diff
@@ -994,16 +994,12 @@ impl<'db> ClassType<'db> {
 pub struct UnionType<'db> {
     /// The union type includes values in any of these types.
     #[return_ref]
-    elements: FxOrderSet<Type<'db>>,
+    elements: Box<[Type<'db>]>,
 }
 
 impl<'db> UnionType<'db> {
-    pub fn contains(&self, db: &'db dyn Db, ty: Type<'db>) -> bool {
-        self.elements(db).contains(&ty)
-    }
```

Comment:
It's not _necessary_ for the change in this PR, but it's related, because we switch from a set to a boxed array, which means `contains` is now `O(n)` instead of `O(1)`, which means we should avoid using it. And we shouldn't need it, either, because generally we should be looking at subtype/assignability, not type identity.

The only uses of this method are related to `Type::Unbound`, and I want to remove that from the type system entirely.

Reply:
Fair. You could also mark it as deprecated and allow its usage. But it's not really important